### DISCUSSION TITLE
check if files in input CSV exists

### DISCUSSION THREAD

**Discussion by hkaspersen** [nextflow-io/nextflow#3444] [Q&A] [answered]
I have a CSV file as input that looks like this:

```
sample,path
sample1,/path/to/sample1.fasta
sample2,/path/to/sample2.fasta
```

I use the following channel factory to handle the input:

```
Channel
    .fromPath(params.input, checkIfExist: true)
    .splitCsv(header:true, sep:",")
    .map { file(it.path) }
    .collect()
```

However, I noticed that if any of the paths to the `fasta` files in the csv does not exist, it runs without throwing an error or warning. Is there a way to implement a `checkIfExist: true` after the `.map` or somewhere, so that nextflow can check if each path exist?

**Comment by mribeirodantas** [comment, marked as answer]
Hello, @hkaspersen! You have a typo in your Nextflow code at the second line. It's `checkIfExists` with an **s** at the end. Having said that, you can use the same option with the `file` function. I have the following file structure in my machine:

```
/Users/mribeirodantas/input.csv
/Users/mribeirodantas/ex2.nf
/Users/mribeirodantas/sampleA
/Users/mribeirodantas/sampleB
```
The content of `input.csv` is:
```
sample,path
sample1,/Users/mribeirodantas/sampleA
sample2,/Users/mribeirodantas/sampleB
sample3,/Users/mribeirodantas/sampleC
```
As you can see, I don't have a `sampleC` file. By running the Nextflow script below (check the option I added to the `file` function),
```
Channel
    .fromPath('input.csv', checkIfExists: true)
    .splitCsv(header:true, sep:",")
    .map { file(it.path, checkIfExists: true) }
    .collect()
    .view()
```
I get what you're looking for:
```
➜  ~ nextflow run ex2.nf
N E X T F L O W  ~  version 22.10.3
Launching `ex2.nf` [nasty_agnesi] DSL2 - revision: 2d1e3edb19
No such file: /Users/mribeirodantas/sampleC

 -- Check script 'ex2.nf' at line: 4 or see '.nextflow.log' file for more details
```

**Reply by hkaspersen** [reply]
Thank you so much! I wasn't aware that the `checkIfExists: true` worked on the `file` function, great!
